compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -0,0 +1,256 @@
+import org.wso2.ballerinalang.compiler.tree.BLangAnnotation;
+import org.wso2.ballerinalang.compiler.tree.BLangEndpoint;
+import org.wso2.ballerinalang.compiler.tree.BLangEnum;
+import org.wso2.ballerinalang.compiler.tree.BLangFunction;
+import org.wso2.ballerinalang.compiler.tree.BLangImportPackage;
+import org.wso2.ballerinalang.compiler.tree.BLangMarkdownDocumentation;
+import org.wso2.ballerinalang.compiler.tree.BLangNode;
+import org.wso2.ballerinalang.compiler.tree.BLangNodeVisitor;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.tree.BLangResource;
+import org.wso2.ballerinalang.compiler.tree.BLangService;
+import org.wso2.ballerinalang.compiler.tree.BLangTypeDefinition;
+import org.wso2.ballerinalang.compiler.tree.BLangVariable;
+import org.wso2.ballerinalang.compiler.tree.BLangXMLNS;
+import org.wso2.ballerinalang.compiler.tree.expressions.BLangMarkdownParameterDocumentation;
+import org.wso2.ballerinalang.compiler.tree.expressions.BLangMarkdownReturnParameterDocumentation;
+import org.wso2.ballerinalang.compiler.tree.types.BLangObjectTypeNode;
+import org.wso2.ballerinalang.compiler.tree.types.BLangRecordTypeNode;
+import org.wso2.ballerinalang.compiler.tree.types.BLangType;
+import org.wso2.ballerinalang.compiler.tree.types.BLangValueType;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.diagnotic.BLangDiagnosticLog;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @since 0.981.0
+ */
+public class DocumentationAnalyzer extends BLangNodeVisitor {
+
+    private static final CompilerContext.Key<DocumentationAnalyzer> DOCUMENTATION_ANALYZER_KEY =
+            new CompilerContext.Key<>();
+
+    private SymbolTable symTable;
+    private SymbolEnv env;
+    private BLangNode parent;
+    private BLangDiagnosticLog dlog;
+
+    public static DocumentationAnalyzer getInstance(CompilerContext context) {
+        DocumentationAnalyzer documentationAnalyzer = context.get(DOCUMENTATION_ANALYZER_KEY);
+        if (documentationAnalyzer == null) {
+            documentationAnalyzer = new DocumentationAnalyzer(context);
+        }
+        return documentationAnalyzer;
+    }
+
+    public DocumentationAnalyzer(CompilerContext context) {
+        context.put(DOCUMENTATION_ANALYZER_KEY, this);
+        this.symTable = SymbolTable.getInstance(context);
+        this.dlog = BLangDiagnosticLog.getInstance(context);
+    }
+
+    public BLangPackage analyze(BLangPackage pkgNode) {
+        parent = pkgNode;
+        SymbolEnv pkgEnv = symTable.pkgEnvMap.get(pkgNode.symbol);
+        pkgNode.topLevelNodes.forEach(topLevelNode -> analyzeNode((BLangNode) topLevelNode, pkgEnv));
+        pkgNode.completedPhases.add(CompilerPhase.CODE_ANALYZE);
+        parent = null;
+        return pkgNode;
+    }
+
+    private void analyzeNode(BLangNode node, SymbolEnv env) {
```

Comment:
I think this `env` and `node.parent` is not used anywhere. So we may not need this `analyzeNode` method at all. Calling `someNode.accept(this)` then and there would be sufficient i guess..